Chocolate_Vending_Machine.java

```diff
@@ -0,0 +1,240 @@
+package Week_7;
+
+import java.util.*;
+
+public class Chocolate_Vending_Machine implements Cash_Inventory, Item_Inventory {
+	private HashMap<Coin, Integer> cash = new HashMap<>();
+	HashMap<Item, Integer> items = new HashMap<>();
+
+	public void initialize_CashInventory() {
+	
+		for (Coin c : Coin.values()) {
+			cash.put(c, 5);
+		}
+	}
+
+	public void initialize_CashInventory(int coins[]) {
+		int i = 0;
+		for (Coin c : Coin.values()) {
+			cash.put(c, coins[i]);
+			i++;
+
+		}
+	}
+
+	public void intialize_Item_Inventory() {
+
+		for (Item it : Item.values()) {
+			items.put(it, 5);
+
+		}
+	}
+
+	public void intialize_Item_Inventory(int item_values[]) {
+		int i = 0;
+		for (Item it : Item.values()) {
+			items.put(it, item_values[i]);
+			i++;
+
+		}
+	}
+
+	public void addCoin(Coin c, int no) {

```

Comment:
For these add/remove functions you should verify that int no > 0

Reply:
So, throwing an error would be a good choice or just returning without doing anything